pymc/model/core.py

```diff
@@ -1645,6 +1650,8 @@ def compile_fn(
                 mode=mode,
                 **kwargs,
             )
+        if borrow_vars:
+            fn.trust_input = True
```

Comment:
This is not related to borrow_vars. Also, no reason to treat any differently than NDarray backend

Reply:
Still not related to borrow_vars

Reply:
Should I drop this then?

Reply:
Yes, I don't think you need any of the changes in `model/core.py` right? The baseclass of the trace is handling all the function creation logic